fili-core/src/test/groovy/com/yahoo/bard/webservice/web/responseprocessors/EtagCacheResponseProcessorSpec.groovy

```diff
@@ -0,0 +1,129 @@
+        json.has(DruidJsonResponseContentKeys.STATUS_CODE.getName()) >> false
+
+        when:
+        etagCacheResponseProcessor.processResponse(json, druidAggregationQuery, Mock(LoggingContext))
+
+        then:
+        1 * httpErrorCallback.dispatch(
+                INTERNAL_SERVER_ERROR_STATUS_CODE,
+                ErrorMessageFormat.INTERNAL_SERVER_ERROR_REASON_PHRASE.format(),
+                ErrorMessageFormat.STATUS_CODE_MISSING_FROM_RESPONSE.format()
+        )
+    }
+
+    def "processResponse reads response from cache when status code is NOT_MODIFIED"() {
+        given:
+        JsonNode json = MAPPER.readTree(
+                String.format(
+                        '{"%s": %d}',
+                        DruidJsonResponseContentKeys.STATUS_CODE.getName(),
+                        NOT_MODIFIED.getStatusCode()
+                )
+        )
+        dataCache.getDataValue(DruidJsonResponseContentKeys.RESPONSE.getName()) >> '[{"k1":"v1"}]'
+
+        when:
+        etagCacheResponseProcessor.processResponse(json, druidAggregationQuery, Mock(LoggingContext))
+
+        then:
+        json.get(DruidJsonResponseContentKeys.RESPONSE.getName()) == MAPPER.readTree('[{"k1":"v1"}]')
+    }
+
+    def "processResponse reports error when status code is OK but etag is missing"() {
+        given:
+        JsonNode json = MAPPER.readTree(
+                String.format(
+                        '{"%s": %d}',
+                        DruidJsonResponseContentKeys.STATUS_CODE.getName(),
+                        OK.getStatusCode()
+                )
+        )
+
+        when:
+        etagCacheResponseProcessor.processResponse(json, druidAggregationQuery, Mock(LoggingContext))
+
+        then:
+        then:
+        1 * httpErrorCallback.dispatch(
+                INTERNAL_SERVER_ERROR_STATUS_CODE,
+                ErrorMessageFormat.INTERNAL_SERVER_ERROR_REASON_PHRASE.format(),
+                ErrorMessageFormat.ETAG_MISSING_FROM_RESPONSE.format()
+        )
+    }
+
+    def "processResponse caches response and etag on OK response"() {
+        given:
+        querySigningService.getSegmentSetId(_ as DruidAggregationQuery) >> Optional.empty()
+        JsonNode json = MAPPER.readTree(
+                String.format(
+                        '''
+                        {
+                            "%s": "someEtag",
+                            "%s": %d
+                        }
+                        ''',
+                        DruidJsonResponseContentKeys.ETAG.getName(),
+                        DruidJsonResponseContentKeys.STATUS_CODE.getName(),
+                        OK.getStatusCode()
+                )
+        )
+
+        when:
+        etagCacheResponseProcessor.processResponse(json, druidAggregationQuery, Mock(LoggingContext))
+
+        then:
+        1 * dataCache.set(DruidJsonResponseContentKeys.RESPONSE.getName(), null, 'null')
+        1 * dataCache.set(DruidJsonResponseContentKeys.ETAG.getName(), null, '"someEtag"')
```

Comment:
This looks wrong... are we really setting the cache value to `"someEtag"`? (ie. including the quotes around the string)